libcfnet/protocol.c

```diff
@@ -88,3 +89,88 @@ Seq *ProtocolOpenDir(AgentConnection *conn, const char *path)
 
     return seq;
 }
+
+bool ProtocolGet(AgentConnection *conn, const char *remote_path,
+                 const char *local_path, const uint32_t file_size, int perms)
+{
+    assert(conn != NULL);
+    assert(remote_path != NULL);
+    assert(local_path != NULL);
+    assert(file_size != 0);
+
+    perms = (perms == 0) ? CF_PERMS_DEFAULT : perms;
+
+    unlink(local_path);
+    FILE *file_ptr = safe_fopen_create_perms(local_path, "w", perms);
```

Comment:
check for return value from unlink()?

Reply:
I'm going to add an `O_EXCL` option for `fopen`. Checking result of `unlink` should be unnecessary then.